### PR TITLE
Blog App: validations, model specs, and n+1 problems

### PR DESCRIPTION
**Pull request by denisdiaconu**
In this project, I added validations to my models, create specs for them, and find and fix N+1 queries problems.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -5,5 +5,6 @@ def index
 
   def show
     @user = User.find(params[:id])
+    @posts = @user.posts
```
Comment (sinansevgi) [inline app/controllers/users_controller.rb:8]: - [x] You should make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
N+1 problem still persists 
Screenshot from 2021-11-06 04-03-46

Tip: You can use the bullet gem to detect N+1 Problems

**Review by sinansevgi — CHANGES_REQUESTED** [review]
# Changes Required :recycle: 
------

Hi @denisdiaconu, 

Good job so far!
Validations and unit tests are perfect :+1: 
There are some issues that you still need to work on to go further but you are almost there!


 ## Required Changes ♻️
_Please check the inline comments for my suggestions_ :arrow_down:

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi

------

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (denisdiaconu) [reply app/controllers/users_controller.rb:8]: Thank you!

**Review by oxenprogrammer — APPROVED** [review]
Hi @denisdiaconu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**